### ISSUE TITLE
'Read timed out' regression in 0.3.1

### ISSUE THREAD

**Issue by CyberBotX** [streamlink/streamlink#550]
This is mainly to say that the issue that was resolved in issue #332 appears to have regressed in 0.3.1, as I am encountering the exceptions once again. This was not happening with 0.3.0 though.

(Issue #547 might be the same thing as well.)

**Comment by beardypig** [comment]
Weird. Is it exactly the same? Can you show me the console output when the error occurs? How did you install streamlink?

**Comment by CyberBotX** [comment]
What I got just moments ago was this:

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 232, in _error_catcher
    yield
  File "/usr/local/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 314, in read
    data = self._fp.read(amt)
  File "/usr/local/lib/python3.5/http/client.py", line 448, in read
    n = self.readinto(b)
  File "/usr/local/lib/python3.5/http/client.py", line 488, in readinto
    n = self.fp.readinto(b)
  File "/usr/local/lib/python3.5/socket.py", line 576, in readinto
    return self._sock.recv_into(b)
  File "/usr/local/lib/python3.5/ssl.py", line 937, in recv_into
    return self.read(nbytes, buffer)
  File "/usr/local/lib/python3.5/ssl.py", line 799, in read
    return self._sslobj.read(len, buffer)
  File "/usr/local/lib/python3.5/ssl.py", line 583, in read
    v = self._sslobj.read(len, buffer)
socket.timeout: The read operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/requests/models.py", line 676, in generate
    for chunk in self.raw.stream(chunk_size, decode_content=True):
  File "/usr/local/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 357, in stream
    data = self.read(amt=amt, decode_content=decode_content)
  File "/usr/local/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 324, in read
    flush_decoder = True
  File "/usr/local/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 237, in _error_catcher
    raise ReadTimeoutError(self._pool, None, 'Read timed out.')
requests.packages.urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='video-edge-835bb4.ord02.hls.ttvnw.net', port=443): Read timed out.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.5/site-packages/streamlink/stream/segmented.py", line 160, in run
    self.write(segment, result)
  File "/usr/local/lib/python3.5/site-packages/streamlink/stream/hls.py", line 120, in write
    for chunk in res.iter_content(chunk_size):
  File "/usr/local/lib/python3.5/site-packages/requests/models.py", line 683, in generate
    raise ConnectionError(e)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='video-edge-835bb4.ord02.hls.ttvnw.net', port=443): Read timed out.
```

As for installing it, I have an in-progress FreeBSD port that I partially based off the old livestreamer port, so it is being installed via the ports system's python handling.

**Comment by beardypig** [comment]
Alright, based on the line number where the exception occurs (`hls.py:120`), I'd say you are using an older version. Can you make sure that you've got the latest version?

**Comment by beardypig** [comment]
@CyberBotX `0.3.2` is out now and has the fix for this issue.

**Comment by CyberBotX** [comment]
Sorry it took me so long to get back on this. As far as I can tell, it appears to be fixed in 0.3.2. But either I am not seeing any read timeout errors coming up or they haven't been happening. I suppose if it comes up again, I could always report it.

**Comment by gravyboat** [comment]
@CyberBotX Let's go with opening a new issue if it shows up again. I'm going to close this for now. Thanks for the update!
